src/ocean/core/SmartUnion.d

```diff
@@ -254,6 +289,33 @@ private template MemberList ( uint i, size_t len, U )
     }
 }
 
+/*******************************************************************************
+
+    Evaluates to a ',' separated list of the names of the members of U, each
+    wrapped with quotes ("").
+
+    Template_Params:
+        i = U member start index
+        U = aggregate to iterate over
+
+    Evaluates to:
+        a ',' separated list of the names of the members of U, each wrapped with
+        quotes ("").
+
+*******************************************************************************/
+
+private template MemberStringList ( uint i, U )
+{
+    static if ( i == U.tupleof.length )
+    {
+        const MemberStringList = "";
+    }
+    else
+    {
+        const MemberStringList = `,"` ~ FieldName!(i, U) ~ `"` ~
+            MemberStringList!(i + 1, U);
+    }
+}
```

Comment:
Suggest generating the actual array of strings here like
```d
// Template code would probably be cumbersome so use CTFE
private static istring[] memberStringList ( U ) ( )
{
    istring[] member_strings = ["none"[]];
    foreach (i, Field; typeof(U.tupleof))
        member_strings ~= FieldName!(i, U);
    return member_strings;
}

// inside SmartUnion
istring[] active_names = memberStringList!(U)();
```
Btw. why did you put `MemberStringList` outside the `SmartUnion` struct?